### PR TITLE
Specify how the user agent should use the approved_clients list

### PR DESCRIPTION
**Pull request by samuelgoto**
***
<a href="https://pr-preview.s3.amazonaws.com/fedidcg/FedCM/pull/270.html" title="Last updated on Jun 7, 2022, 5:03 AM UTC (858a7a6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/fedidcg/FedCM/270/244c4c6...858a7a6.html" title="Last updated on Jun 7, 2022, 5:03 AM UTC (858a7a6)">Diff</a>

### CODE REVIEW

**Review by npm1 — APPROVED** [review]
Looks good. Perhaps we can add some text explaining this in the description in section 7.2

**Comment by samuelgoto** [comment]
> Looks good. Perhaps we can add some text explaining this in the description in section 7.2

Good point! Done!

**Comment by samuelgoto** [comment]
Neat, merging!

File: spec/index.bs [hunk 1]
```diff
@@ -1185,13 +1185,15 @@ To <dfn>fetch the accounts list</dfn>:
...
-To <dfn>sign-up</dfn> the user:
+To <dfn>sign-up</dfn> the user with a given |account|:
     1. Let |metadata| be the result of running the [=fetch the client metadata=]
         algorithm.
-    1. If |metadata|["privacy_policy_url"] is defined then display the
-        |metadata|["privacy_policy_url"] link.
-    1. If |metadata|["terms_of_service_url"] is defined then display the
-        |metadata|["terms_of_service_url"] link.
+    1. If |metadata|["{{client_metadata_endpoint_response/privacy_policy_url}}"] is defined and
+        the provider["{{FederatedIdentityProvider/clientId}}"] is not in the list of |account|["{{accounts_endpoint_response_accounts/approved_clients}}"]
```
Comment (cbiesinger) [inline spec/index.bs:1192]: As written, this assumes that the IDP did send an approved_clients list, but that is optional. Should probably talk about what happens if that list was not sent?

Comment (samuelgoto) [reply spec/index.bs:1192]: Doesn't this already address the case where it is not provided? Meaning, that "if" never executes, so it goes on to do nothing? Is there anything that gets done when this optional field is NOT passed that I forgot to capture?

Comment (cbiesinger) [reply spec/index.bs:1192]: The if is about privacy_policy_url, not about approved_clients?

Comment (samuelgoto) [reply spec/index.bs:1192]: Ah, good point, will send a PR to resolve that!

**Pull request merged**